### PR TITLE
FOGL-9129: North complex pipeline does not allow to choose a filter

### PR DESCRIPTION
**Pull request by hemantgarg923**

### CODE REVIEW

File: src/app/components/common/node-editor/editor.ts [hunk 1]
```diff
@@ -335,25 +335,29 @@ export function getUpdatedFilterPipeline() {
...
     else {
+      // check if all the filters are connected from both sides
       if (!connections.find(c => c.source === nodes[i].id) || !connections.find(c => c.target === nodes[i].id)) {
         console.log("Dangling connection");
         return false;
       }
     }
   }
 
+  // check for self loop
```
Comment (praveen-garg) [inline src/app/components/common/node-editor/editor.ts:360]: self loop?

**Review by praveen-garg — APPROVED** [review]

File: src/app/components/common/node-editor/editor.ts (hunk 1 above)
Comment (hemantgarg923) [reply src/app/components/common/node-editor/editor.ts:360]: This check is to test if there is a connection from input socket to output socket of same node.

File: src/app/components/common/node-editor/insert-node/index.ts [hunk 2]
```diff
@@ -94,8 +94,10 @@ async function removeOldConnection(node, editor) {
       inputConnId = element.id;
     }
   }
-  for (let t of target) {
-    await editor.addConnection(new Connection(connectionEvents, source, t));
+  if(source) {
+    for (let t of target) {
```
Comment (MonikaSharma06) [inline src/app/components/common/node-editor/insert-node/index.ts:98]: It should be like-

`for (let target of targets) {`

**Review by MonikaSharma06 — DISMISSED** [review]

File: src/app/components/common/node-editor/insert-node/index.ts (hunk 2 above)
Comment (praveen-garg) [reply src/app/components/common/node-editor/insert-node/index.ts:98]: agree `targets` as plural!

`for (let target of targets) {`  or `for (let t of targets) {`

File: src/app/components/common/node-editor/editor.ts [hunk 3]
```diff
@@ -335,25 +335,29 @@ export function getUpdatedFilterPipeline() {
 
   for (let i = 0; i < nodes.length; i++) {
     if (i == 0) {
+      // check if starting node of pipeline i.e. south/storage is connected
       if (!connections.find(c => c.source === nodes[i].id)) {
         console.log("Dangling connection");
```
Comment (praveen-garg) [inline src/app/components/common/node-editor/editor.ts:340]: not in the scope of this, but should we log more details in logs? Dangling connection for or from .. to ...

**Review by praveen-garg — DISMISSED** [review]

**Review by mshariq-nerd — APPROVED** [review]

**Review by mshariq-nerd — APPROVED** [review]

**Pull request merged**